### ISSUE TITLE
How to make sure your pre-trained model from other works has the same performance with others?

### ISSUE THREAD

**Issue by StefanIsSmart** [jiaor17/3D-MGP#1]

**Comment by jiaor17** [comment]
Hi @StefanIsSmart,
Thank you for your interest! In this work, we adapt several baseline methods previously used for 2D GNNs and apply them to 3D GNNs (such as EGNN) to assess their effectiveness in this new context.  The re-implementation codes are provided in this [folder](https://github.com/jiaor17/3D-EMGP/tree/main/mgp/baselines), which follows the previous repos like [pretrain-gnns](https://github.com/snap-stanford/pretrain-gnns) and [GraphMVP](https://github.com/chao1224/GraphMVP).

**Comment by StefanIsSmart** [comment]
Thank you for your reply.

How to plot the embedding of different pre-trained models? 
Could you give a jupyter script?
Because I find the input data of different models are different, even finetuning has different input with pre-training.

And why use the same learning rate with the pre-trained encoder and new predictor heads?

**Comment by jiaor17** [comment]
Hi,
Sorry that the visualization codes might have been removed, but you can refer to Section 4.4 of the paper to plot the landscape. We use the same learning rate as the encoder and the prediction heads are jointly considered as the predictor to acquire the results. Regarding the different input data, could you please specify the problem and give some examples? This will help me assist you further.

**Comment by StefanIsSmart** [comment]
For example, the edge_attr are different between md17 and qm9:

- md17:
```
 edge_attr = nn.functional.one_hot(data.edge_type, 4).to(device,dtype)
```

- qm9:
```
        if config.model.no_edge_types:
            edges, edge_types = gen_fully_connected(atom_positions, atom_mask)
            edge_attr = None
        else:
            edges, edge_types = gen_fully_connected_with_hop(atom_positions, atom_mask)
            edge_attr = nn.functional.one_hot(edge_types, 4)
```


How to make sure the input is right when I want to get the embedding of checkpoints from different models?
(By the way, the embedding should be generated by the `EGNN_last` model right?

**Comment by jiaor17** [comment]
Hi,
Thanks for bringing this to our attention! In our implementation, the edge features are categorized into a 4-dimensional vector representing [1-hop, 2-hop, 3-hop, others] as described in Appendix B.1 of our [associated paper](https://arxiv.org/pdf/2207.08824). The 1-hop connections are identified based on covalent bonds. If the molecular data you're working with already includes bond information, you can utilize that directly to determine the 1-hop connections. If not, we recommend using a distance cutoff of 1.6 Angstroms, based on common chemical knowledge, to infer likely covalent bonds.
Regarding the embeddings, they are generated by the `EGNN_last` encoder, and one can also extract the scaler output of the prediction head after finetuning for specific applications.
Hope the above information could help you!

**Comment by StefanIsSmart** [comment]
Thank you again!
It is helpful.

So I could get a `mol` from smiles with RDKit, and get the `chemical bond type` -> edge_type, then just transform it by `nn.functional.one_hot(data.edge_type, 4)`

When inferencing with encoder, I needn't add noise to the position like training.

Right?

**Comment by jiaor17** [comment]
Hi,
In this work, the `edge_attr` is defined by the number of hops, rather than specific chemical bonds. You can obtain the hops using the [EdgeHop class](https://github.com/jiaor17/3D-EMGP/blob/main/data/dataset.py#L456) after constructing the `edge_index` with RDKit. Additionally, adding noise during inference is not necessary.

**Comment by StefanIsSmart** [comment]
Great work!
Thank you !

**Comment by StefanIsSmart** [comment]
Great work! Thank you!

**Comment by StefanIsSmart** [comment]
Hello!
One more thing! Could you update the pre-training config file of other baselines?

**Comment by StefanIsSmart** [comment]
And where are the checkpoints of JOAO, JOAOv2 and GCC?

**Comment by StefanIsSmart** [comment]
@jiaor17  :p

**Comment by jiaor17** [comment]
Hi,

Sorry for the late reply. The config file is actually a mixed configuration that includes the hyperparameters used for all baselines. You can use it directly to run the baselines. Additionally, the scripts and checkpoints for the contrastive baselines have now been updated.
